File: tic-tac-toe-core/src/main/java/by/xxx/pupil/ai/minimax/findmoves/AllAvailableMovesFinder.java
package by.xxx.pupil.ai.minimax.findmoves;

import by.xxx.pupil.model.Board;
import by.xxx.pupil.model.Move;
import by.xxx.pupil.model.Player;

import java.util.ArrayList;
import java.util.List;

import static by.xxx.pupil.BoardUtils.isCellEmpty;

public class AllAvailableMovesFinder implements MovesFinder {

    @Override
    public List<Move> getMoves(Board board, Player player) {
        List<Move> availableMoves = new ArrayList<>();
        for (int i = 0; i < board.getHeight(); i++) {
            for (int j = 0; j < board.getWidth(); j++) {
                if (isCellEmpty(board, i, j)) {
                    availableMoves.add(new Move(i, j, player));
                }
            }
        }

        return availableMoves;
    }
}

File: tic-tac-toe-core/src/main/kotlin/by/xxx/pupil/ai/minimax/findmoves/AllAvailableMovesFinder.kt
package by.xxx.pupil.ai.minimax.findmoves

import by.xxx.pupil.model.Board
import by.xxx.pupil.model.Move
import by.xxx.pupil.model.Player
import by.xxx.pupil.model.isCellEmpty
import java.util.*

class AllAvailableMovesFinder : MovesFinder {
    override fun getMoves(board: Board, player: Player): List<Move> {
        val availableMoves: MutableList<Move> = ArrayList()
        for (i in 0 until board.height) {
            for (j in 0 until board.width) {
                if (isCellEmpty(board, i, j)) {
                    availableMoves.add(Move(i, j, player))
                }
            }
        }
        return availableMoves
    }
}